sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -198,8 +200,9 @@ abstract class QueryGenerator(
         // Replace the named argument with a non named/indexed argument.
         // This allows us to use the same algorithm for non Sqlite dialects
         // :name becomes ?
-        if (bindArg != null) {
-          replacements.add(bindArg.range to "?")
```

Comment:
Do you still want to replace all parameters with `?` in the generator, even if the dialect supports/requires other formats, eg `$1` or `:id`? Otherwise I would use the specified value from the dialect/grammar.